python/paddle/v2/dataset/common.py

```diff
@@ -201,8 +201,10 @@ def close_writers(w):
     def write_data(w, lines):
         random.shuffle(lines)
         for i, d in enumerate(lines):
-            d = pickle.dumps(d, pickle.HIGHEST_PROTOCOL)
```

Comment:
I modify the protocol version from *HIGHEST_PROTOCOL* to 0, because this will cause an `EOFError`:
```bash
Traceback (most recent call last):
  File "decode.py", line 13, in <module>
    print pickle.loads(o)
EOFError
```

I think this error maybe caused by RecordIO library: **Use binary stream as input will caused an EOFError**, the following is my test code, it will throw an EOFError
```python
w = recordio.writer("tmp-recordio")
for i in xrange(10):
    w.write(pickle.dumps(i, pickle.HIGHEST_PROTOCOL))
w.close()

r = recordio.reader("tmp-recordio")
for i in xrange(10):
    o = r.read()
    print pickle.loads(o)
r.close()
=========
Traceback (most recent call last):
  File "decode.py", line 13, in <module>
    print pickle.loads(o)
EOFError
```

I will follow this problem,  and maybe do not block this fix:)

Some  documents:
- [About pickle protocol](https://docs.python.org/2/library/pickle.html#data-stream-format)
- [Python pickle protocol choice?](https://stackoverflow.com/questions/23582489/python-pickle-protocol-choice)